compiler/ir/backend.jvm/lower/src/org/jetbrains/kotlin/backend/jvm/lower/JvmInlineClassLowering.kt

```diff
@@ -569,6 +569,9 @@ private class JvmInlineClassLowering(context: JvmBackendContext) : JvmValueClass
 
         val oldBody = function.body
 
+        fun IrClass.isFunctionDeclared(): Boolean =
+            declarations.any { it is IrSimpleFunction && it.name.asString() == name && !it.isFakeOverride }
```

Comment:
What about user defined functions with name clashes?
```kotlin
sealed value class A {
  fun `toString-impl`(x: Int) = x
}
value class B : A()
```

Reply:
Come to think of it, this would already fail with an `IllegalStateException` from `irClass.declarations.single` in line 568.